enterprise/server/backends/distributed/distributed.go

```diff
@@ -266,6 +301,96 @@ func (c *Cache) Check(ctx context.Context) error {
 	return nil
 }
 
+type teeReadCloser struct {
+	rc  io.ReadCloser
+	cwc interfaces.CommittedWriteCloser
+}
+
+func (t *teeReadCloser) Read(p []byte) (n int, err error) {
+	n, err = t.rc.Read(p)
+	if n > 0 {
+		if n, err := t.cwc.Write(p[:n]); err != nil {
+			return n, err
+		}
+	}
+	return
+}
+func (t *teeReadCloser) Close() error {
+	err := t.rc.Close()
+	if err == nil {
+		_ = t.cwc.Commit()
+		_ = t.cwc.Close()
+	}
+	return err
+}
+
+func (c *Cache) addLookasideEntry(r *rspb.ResourceName, data []byte) {
+	if !c.lookasideCacheEnabled() {
+		return
+	}
+	k, err := digest.ResourceNameFromProto(r).DownloadString()
+	if err != nil {
+		c.log.Debugf("Not setting lookaside entry: %s", err)
+		return
+	}
+	entry := lookasideCacheEntry{
+		expiresAfterNanos: time.Now().Add(*lookasideCacheTTL).UnixNano(),
+		data:              data,
+	}
+
+	c.lookasideMu.Lock()
+	c.lookaside.Add(k, entry)
+	c.lookasideMu.Unlock()
+	c.log.Debugf("Set %q in lookaside cache", k)
+}
+
+func (c *Cache) getLookasideEntry(r *rspb.ResourceName) ([]byte, error) {
+	if !c.lookasideCacheEnabled() {
+		return nil, status.NotFoundError("lookaside cache disabled")
+	}
+	k, err := digest.ResourceNameFromProto(r).DownloadString()
+	if err != nil {
+		c.log.Debugf("Not getting lookaside entry: %s", err)
+		return nil, err
+	}
+	c.lookasideMu.RLock()
+	entry, ok := c.lookaside.Get(k)
+	c.lookasideMu.RUnlock()
```

Comment:
`Get` does a mutate of the LRU so this might need to be Lock? (maybe worth also adding a concurrency test since the race detector didn't trigger)

Reply:
Fixed.